flexget/plugins/operate/free_space.py

```diff
@@ -1,19 +1,65 @@
 import shutil
-from pathlib import Path
+from enum import Enum
+from pathlib import Path, PurePath
+from typing import Annotated, TypeAlias
 
+from datamodel_code_generator.util import model_validator
 from loguru import logger
+from pydantic import BaseModel, BeforeValidator, ByteSize, ConfigDict, Field
 
 from flexget import plugin
-from flexget.config_schema import parse_size
 from flexget.event import event
 
 logger = logger.bind(name='free_space')
 
-ABORT_BELOW = 'below'
-ABORT_ABOVE = 'above'
 
+class AbortMode(str, Enum):
+    BELOW = 'below'
+    ABOVE = 'above'
 
-def get_free_space(config, task):
+
+# Original config had these fields as MiB when a raw number was supplied, but we want bytes now
+def convert_to_bytes(val):
+    if isinstance(val, (int, float)):
+        return val * 1024 * 1024
+    return val
+
+
+BackCompatByteSize: TypeAlias = Annotated[
+    ByteSize,
+    Field(json_schema_extra=lambda s: s['anyOf'].append({'type': 'number'})),
+    BeforeValidator(convert_to_bytes),
+]
```

Comment:
I think probably the route to go for this specific change would be to drop backwards compatibility and require all sizes declare their unit. I sorta like that it's possible to describe exactly what we want here though. Schema allows numbers or string with unit, but the parsed config will only be one type. No need to handle the case separate after validation.